controllers/snapshot/snapshot_adapter.go

```diff
@@ -477,7 +477,19 @@ func (a *Adapter) ensureEphemeralEnvironmentForScenarioExists(integrationTestSce
 	environment, ok := a.getExistingEnvironmentForScenario(integrationTestScenario, allEnvironments)
 	if !ok {
 		// environment doesn't exist, we have to create a new one
-		environment, err = a.createEnvironmentForScenario(integrationTestScenario)
+		existingEnv := findEnvironmentFromAllEnvironmentsByName(allEnvironments, integrationTestScenario.Spec.Environment.Name)
+		if existingEnv == nil {
+			a.logger.Error(fmt.Errorf("environment doesn't exist in the same namespace as integrationScenario at all, try again after creating environment"),
```

Comment:
https://pkg.go.dev/github.com/go-logr/logr@v1.2.4#Logger.Error

it doesn't match signature of function, `msg` is missing. I think we can pass `nil` as `err` here

Reply:
Yes, need a msg. Thanks!

Reply:
Fixed.